src/ui/public/courier/data_source/search_source.js

```diff
@@ -268,6 +269,9 @@ export function SearchSourceProvider(Promise, Private, config) {
         val = normalizeSortRequest(val, this.get('index'));
         addToBody();
         break;
+      case 'fields':
+        state[key] = [...state[key] || [], ...val];
```

Comment:
If `state[key]` is empty, won't this result in an array that looks like this:

```
[[], 'stuff', 'from', 'val']
```

Or does `||` take precedence over `...`? If it does, mind adding parenthesis to make it more obvious?

Might want to dedupe identical strings as well?